registration/include/pcl/registration/impl/ndt.hpp

```diff
@@ -419,7 +412,9 @@ NormalDistributionsTransform<PointSource, PointTarget>::computeHessian (Eigen::M
   // Update hessian for each point, line 17 in Algorithm 2 [Magnusson 2009]
   for (std::size_t idx = 0; idx < input_->points.size (); idx++)
   {
-    x_trans_pt = trans_cloud.points[idx];
+  // Transformed Point
+    PointSource x_trans_pt = trans_cloud.points[idx];
```

Comment:
Check if you can simply take a reference instead of making a copy. Also always prefer `trans_cloud[idx]`, we might hide visibility to `point` in the future.

Reply:
Unnecessary copies are replaced with const references.